eclair-core/src/test/scala/fr/acinq/eclair/integration/OfferIntegrationSpec.scala

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2023 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.integration
+
+import akka.actor.testkit.typed.scaladsl.{TestProbe => TypedProbe}
+import akka.actor.typed.scaladsl.adapter._
+import akka.pattern.pipe
+import akka.testkit.TestProbe
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
+import fr.acinq.bitcoin.scalacompat.{Crypto, SatoshiLong}
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher.{Watch, WatchFundingConfirmed}
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.db.OutgoingPayment
+import fr.acinq.eclair.message.OnionMessages.{IntermediateNode, Recipient, buildRoute}
+import fr.acinq.eclair.offer.OfferManager._
+import fr.acinq.eclair.payment.receive.MultiPartHandler.{DummyBlindedHop, ReceivingRoute}
+import fr.acinq.eclair.payment.send.PaymentIdentifier
+import fr.acinq.eclair.payment.{Bolt12Invoice, PaymentReceived, PaymentSent}
+import fr.acinq.eclair.router.Router
+import fr.acinq.eclair.wire.protocol.OfferTypes._
+import fr.acinq.eclair.{CltvExpiryDelta, EclairImpl, Features, Kit, MilliSatoshiLong, randomBytes32, randomKey}
+import scodec.bits.ByteVector
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters._
+
+class OfferIntegrationSpec extends IntegrationSpec {
```

Comment:
You don't need a real `bitcoind` for those tests, it makes these tests much slower than they need to be. You should use the `MinimalNodeFixture` framework instead of spinning up complete eclair nodes.

Reply:
I've moved all the tests to `BlindedPaymentSpec` and `PaymentIntegrationSpec`.